### PR TITLE
[IMPROVE] Use MongoBD aggregation to get users from a room

### PR DESCRIPTION
**Pull request by ggazzo**
Closes #ISSUE_NUMBER

### CODE REVIEW

File: server/methods/getUsersOfRoom.js [hunk 1]
```diff
@@ -16,17 +16,32 @@ Meteor.methods({
...
 
 		return {
-			total: userIds.length,
-			records: users,
+			total: subscriptions.count(),
+			records: await RocketChat.models.Subscriptions.model.rawCollection().aggregate([
+				{ $match: { rid } },
+
+				{
+					$lookup:
```
Comment (sampaiodiego) [inline server/methods/getUsersOfRoom.js]: mongodb applies the same [BSON Document Size](https://docs.mongodb.com/manual/reference/limits/#BSON-Document-Size) limit to lookup results.. it didn't reach this limit on my 200k users database, but I'm almost sure that limit will be hit on larger databases which will make this query to fail

Comment (ggazzo) [reply server/methods/getUsersOfRoom.js]: 🙄

Comment (ggazzo) [reply server/methods/getUsersOfRoom.js]: @sampaiodiego are you sure about that? I made a fast search here:
https://docs.mongodb.com/manual/core/aggregation-pipeline-limits/#result-size-restrictions
and if I'm not wrong the limit just applies for results(16mb per item), I dont think your subscription + user will get that size, and we can use `$project` to reduce the size...could you check it again? <3

Comment (engelgabriel) [reply server/methods/getUsersOfRoom.js]: *EACH* document in the result set is subject to the BSON Document Size limit, currently 16 megabytes;

File: server/methods/getUsersOfRoom.js [hunk 2]
```diff
@@ -16,17 +16,32 @@ Meteor.methods({
...
+				},
+				{
+					$project: {
+						'u._id': 1,
+						'u.name': 1,
+						'u.username': 1,
+						'u.status': 1,
+					},
+				},
+				...(showAll ? [{ $match: { 'u.status': 'online' } }] : []),
```
Comment (sampaiodiego) [inline server/methods/getUsersOfRoom.js]: the test is inverted :stuck_out_tongue:

**Comment by ggazzo** [comment]
I think its good now

**Comment by wreiske** [comment]
Maybe this is why my member's list isn't working? 
@engelgabriel @sampaiodiego @ggazzo 
screen shot 2018-12-27 at 10 12 24 pm
```Exception in delivering result of invoking 'getUsersOfRoom': TypeError: Cannot read property 'records' of undefined
    at o.call (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:467)
    at n._callback (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:1)
    at n._maybeInvokeCallback (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:61)
    at n.receiveResult (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:61)
    at w._livedata_result (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:61)
    at w.onMessage (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:61)
    at currentStatus.connected.forEachCallback.e (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:59)
    at Array.forEach (<anonymous>)
    at l.forEachCallback (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:59)
    at k.socket.onmessage.t [as onmessage] (3e934eeee51ba52a4c3e836911d39392cdb5a05f.js:59)```
![screen shot 2018-12-27 at 10 15 02 pm](https://user-images.githubusercontent.com/6295044/50501262-e6df3900-0a24-11e9-8748-ab2e8e69b76c.png)

**Comment by geekgonecrazy** [comment]
Either replaceRoot has got to go... or we need big disclaimer that mongo 3.4 is required.

https://docs.mongodb.com/manual/reference/operator/aggregation/replaceRoot/

> new in 3.4 

@ggazzo @engelgabriel @sampaiodiego @rodrigok 

If we force 3.4 I believe all arm builds go out the door as a possibility.  At least any 32bit ones

**Comment by engelgabriel** [comment]
This is a tricky one because we are also planning to use Change Streams (New in version 3.6) for high performance. Can we please check this restriction about arm builds?

**Pull request merged**